src/test/java/org/ice4j/ice/NetworkUtilsTest.java

```diff
@@ -42,6 +76,12 @@ public void testIpv6StringToBytes()
         assertNotNull(addr);
         assertEquals(18, addr[15]);
 
+        assertNull(NetworkUtils.strToIPv6(""));
+        assertNull(NetworkUtils.strToIPv6(":::"));
+        assertNull(NetworkUtils.strToIPv6("[^"));
+        assertNull(NetworkUtils.strToIPv6("[%"));
+        assertNull(NetworkUtils.strToIPv6(":?0"));
         assertNull(NetworkUtils.strToIPv6("[::12]%1"));
+        assertNull(NetworkUtils.strToIPv6("[::65536]%1"));
```

Comment:
I'd be interested also to see some tests that describe how this method handles short form notation for IPv6 addresses. 

>  An IPv6 (Normal) address has the following format: y : y : y : y : y : y : y : y where y is called a segment and can be any hexadecimal value between 0 and FFFF. The segments are separated by colons - not periods. An IPv6 normal address must have eight segments, however a short form notation can be used in the Tape Library Specialist Web interface for segments that are zero, or those that have leading zeros. The short form notation can not be used from the operator panel.
> The following list shows examples of valid IPv6 (Normal) addresses:
> 
>     2001 : db8: 3333 : 4444 : 5555 : 6666 : 7777 : 8888
>     2001 : db8 : 3333 : 4444 : CCCC : DDDD : EEEE : FFFF
>     : : (implies all 8 segments are zero)
>     2001: db8: : (implies that the last six segments are zero)
>     : : 1234 : 5678 (implies that the first six segments are zero)
>     2001 : db8: : 1234 : 5678 (implies that the middle four segments are zero)
>     2001:0db8:0001:0000:0000:0ab9:C0A8:0102 (This can be compressed to eliminate leading zeros, as follows: 2001:db8:1::ab9:C0A8:102 )

Reply:
Thanks for that. There are some short form IPv6 addresses included in existing + added tests eg. "::12"